src/librustdoc/html/static/js/search.js

```diff
@@ -1613,7 +1634,19 @@ window.initSearch = function(rawSearchIndex) {
     }
 
     function updateCrate(ev) {
-        updateLocalStorage("rustdoc-saved-filter-crate", ev.target.value);
+        if (ev.target.value === "All crates") {
+            removeLocalStorageValue("saved-filter-crate");
+            // If we don't remove it from the URL, it'll be picked up again by the search.
+            var params = searchState.getQueryStringParams();
+            var query = searchState.input.value.trim();
```

Comment:
Why is the "update URL" code path only activated when selecting "All crates"? It seems like we should always update the URL after the user makes a selection from the crates dropdown.

Reply:
Because it'll be done in the search for the other cases (ie, when we have a filter crate). This needs to be changed the current URL so that the filter crate won't be picked since there isn't one anymore.